webapp/src/components/ProgramActivityDialog.tsx

```diff
@@ -138,16 +272,69 @@ const ProgramActivityDialog = ({
           </Table>
         </TableContainer>
       </DialogContent>
-
-      <Divider />
-
-      <DialogActions>
-        <Button onClick={handleClose} variant="contained">
-          Close
-        </Button>
-      </DialogActions>
+      {contents.activityType === 'assignment' && (
+        <>
+          <Divider />
+          <DialogActions>
+            <FormGroup
+              sx={{
+                display: 'flex',
+                flexDirection: 'row',
+                justifyContent: 'flex-end',
+              }}
+              id="form"
+            >
+              <LoadingButton
+                onClick={event => {
+                  sendCompletedStatus(event, !completed);
+                }}
+                type="submit"
+                form="form"
+                variant={completed === false ? 'contained' : 'outlined'}
+                loading={isLoading}
+                sx={{ width: '15em' }}
+                disabled={error ? true : false}
+              >
+                {completed === false ? (
+                  <TaskAltIcon sx={{ mr: 1 }} />
+                ) : (
+                  <CancelIcon sx={{ mr: 1 }} />
+                )}
+                {completed === false ? 'Mark Complete' : 'Mark Incomplete'}
+              </LoadingButton>
+              {isMessageVisible && (
+                <Snackbar
+                  open={true}
+                  autoHideDuration={1500}
+                  onClose={() => setIsMessageVisible(false)}
+                >
+                  <Alert
+                    onClose={() => setIsMessageVisible(false)}
+                    severity={isUpdateSuccess ? 'success' : 'error'}
+                    sx={{ width: '100%' }}
+                  >
+                    {isUpdateSuccess
+                      ? 'Assignment status updated successfully.'
+                      : 'There was an error updating the assignment status.'}
+                  </Alert>
+                </Snackbar>
+              )}
+              {isErrorShown && (
```

Comment:
In general, <abbr>DRY</abbr> (don't repeat yourself) whenever possible when an overload of existing functionality could easily work, but I think in this case it's fine because we're going to remove lines 322 on when we consolidate loading the activity statuses all at once, so this is okay.